src/components/calculator.js

```diff
@@ -1,43 +1,52 @@
-import React from 'react';
+import { useState } from 'react';
 import Button from './button';
+import calculate from '../logic/calculate';
+
+export default function Calculator() {
+  const [state, setState] = useState({
+    next: null,
+    operation: null,
+    total: null,
+  });
+  const { next, operation, total } = state;
+  const eventHandler = (e) => {
+    setState(calculate(state, e.target.name));
+  };
 
-export default function calculator() {
   return (
     <div className="calculator-container">
       <div className="calculator">
         <div className="row">
-
-          <input type="number" placeholder="0" />
-
+          <div className="display">{next || operation || total || 0}</div>
         </div>
         <div className="row">
-          <Button className="column" value="AC" />
-          <div className="column">+/-</div>
-          <div className="column">%</div>
-          <div className="column orange">/</div>
+          <Button className="column" value="AC" handleClick={eventHandler} />
+          <Button className="column" value="+/-" handleClick={eventHandler} />
+          <Button className="column" value="%" handleClick={eventHandler} />
+          <Button className="column orange" value="/" handleClick={eventHandler} />
         </div>
         <div className="row">
-          <div className="column">7</div>
-          <div className="column">8</div>
-          <div className="column">9</div>
-          <div className="column orange">*</div>
+          <Button className="column" value="7" handleClick={eventHandler} />
+          <Button className="column" value="8" handleClick={eventHandler} />
+          <Button className="column" value="9" handleClick={eventHandler} />
+          <Button className="column orange" value="*" handleClick={eventHandler} />
         </div>
```

Comment:
- [ ] The calculate function uses the `x` instead of the `*` symbol. Therefore, when I try to multiply two numbers, I get the error in the screenshot below:

![image](https://github.com/isume295/maths/assets/77106962/052cd39f-367b-4d61-9d90-2e7ae8f0bcc6)

Kindly use the  `x` symbol. 👍 💻 

#### Hint: You can get the same symbol from the `src/logic/operate.js` file. 😎